---
title: "PCOS-Androgen-related metabolic disturbances (Data Analysis w/ GEO Query)"
output: html_notebook
---

The purpose of this notebook is to explore data from the study: 	
Liao B, Qi X, Yun C, Qiao J et al. Effects of Androgen Excess-Related Metabolic Disturbances on Granulosa Cell Function and Follicular Development. Front Endocrinol (Lausanne) 2022;13:815968. PMID: 35237237
The data was obtained using GEOQuery. The goal is to explore the regulation of genes related to both anrogen and estrogen function. The GEO acession number for this study is GSE193123. This analysis is a preliminary analysis before conducting a full differential gene expression analysis using this same dataset. 
```{r}
library(dplyr)
library(tidyverse)
library(GEOquery)
```
```{r}
BiocManager::install("EnsDb.Hsapiens.v79")
```

Here I read in the dataframe downloaded from GEO as well as the metadata for this dataset. 
```{r}
df<- read.delim( file = "/Users/danielaquijano/Documents/GitHub/RNA-Seq-Workflows/GSE193123_gene_fpkm.txt",sep="\t")
dim(df)
```
```{r}
head(df)
```
```{r}
gse <- getGEO(GEO = 'GSE193123', GSEMatrix = TRUE)
```
```{r}
gse
```
```{r}
#GEO Series are collections of related experiments, parse data with pData()
metadata <- pData(phenoData(gse[[1]])) #First object of the first element in the list. 
head(metadata)
```
With the metadata information we can see that we have six samples. Three come from healthy women with no PCOS and the other samples come from women with PCOS.


```{r}
#Select columns from metadata and label samples and corresponding diagnosis
metadata.modified <- metadata %>%
  select(1,2,12,17) %>% 
  rename(diagnosis= characteristics_ch1.2) %>%
  rename( sample=title ) %>%
  mutate(diagnosis = gsub("diagnosis: ", "", diagnosis)) 
```
```{r}
df
```
```{r}
df_only_counts=df  %>%
  select(2,3,4,5,6,7,8) 
#rownames(df_only_counts)<-df$gene_id
```

```{r}
#Reshape gene expression dataframe (df) to long 
data_long <- gather(df_only_counts, key='samples', value='FPKM', C1:P3, factor_key=TRUE)
data_long
```
```{r}
data_long_joined <- data_long %>%
  left_join(., metadata.modified, by = c("samples" = "description")) 
```
```{r}
head(data_long_joined)
```
Based on literature searchers, relevant genes
Literature used:
https://www.ncbi.nlm.nih.gov/pmc/articles/PMC3356164/
```{r}
genes_list<-c('AR','ERA','CYP19A1','SRD5A2','NR3A2','CYP17','SHBG','ESR1','UGT1A1','PGR','COMT','CYP1A1','CYP1B1','CYP17A1','CYP19A1','GSTM1','GSTP1','GSTT1','HSD17B1','SULT1A1','UGT1A1')
```

```{r}
data_long_joined %>% filter(gene_name %in% genes_list) %>%
  group_by(gene_name,diagnosis) %>%
  summarize(mean_FPKM = mean(FPKM), median_FPKM = median(FPKM)) %>%
  arrange(-mean_FPKM)
```
```{r}
data_long_joined  %>%
  filter(gene_name %in% genes_list) %>%
  ggplot(., aes(x = samples, y = gene_name, fill = FPKM)) +
  geom_tile() +
  scale_fill_gradient( low = "blue", high = "pink")
```
```{r}
#Grouped bar-graph
data_long_joined %>%
  filter(gene_name == 'CYP19A1') %>%
  ggplot(., aes(x = samples, y = FPKM, fill = diagnosis)) +
  geom_col()
```
```{r}
#data_long_joined %>%
  #filter(gene_name=='CYP19A1') %>%
  #ggplot(., aes(x = FPKM, fill = diagnosis)) +
  #geom_density(alpha = 0.3)
```
```{r}

